promql/engine.go

```diff
@@ -3277,6 +3277,17 @@ func (ev *evaluator) aggregationK(e *parser.AggregateExpr, fParam float64, input
 		}
 	}
 
+	switch op {
+	case parser.TOPK, parser.BOTTOMK, parser.LIMITK:
+		if !convertibleToInt64(fParam) {
+			ev.errorf("Scalar value %v overflows int64", fParam)
+		}
+	case parser.LIMIT_RATIO:
+		if math.IsNaN(fParam) {
```

Comment:
I think we can just check for a `if params.HasAnyNaN() {` here https://github.com/prometheus/prometheus/blob/4aee718013c52bef1ccb336bcc21886336e6252e/promql/engine.go#L1438

Reply:
done